File: Data Structures & Algorithms/transpose-matrix/submission-1.py
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        rows,cols = len(matrix), len(matrix[0])
        t = [[0] * rows for _ in range(cols)]
        for r in range(rows):
            for c in range(cols):
                t[c][r] = matrix[r][c]
        return t

File: Data Structures & Algorithms/rotate-matrix/submission-1.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int n = matrix.size();
        for (int r= 0; r<n;r++) {
            for (int c= 0; c<r; c++) {
                swap(matrix[r][c], matrix[c][r]);
            }
        }
        for (int row = 0; row<n; row++) {
            int l = 0, r= n-1;
            while(l<r) {
                swap(matrix[row][l], matrix[row][r]);
                r--;
                l++;
            }
        }
        
        
    }
};
